reqwest-middleware 0.2 breaks RVCR

So I got a bit further this time but it seems there is some API change in `reqwest-middleware` version `0.2` that breaks RVCR. I got further with `0.1.6` but on `0.2` onwards I get the following type error when trying to follow the example:

code:

```rust
  let middleware: VCRMiddleware = VCRMiddleware::try_from(bundle.clone())
      .unwrap()
      .with_mode(VCRMode::Record);

  let vcr_client: ClientWithMiddleware = ClientBuilder::new(reqwest::Client::new())
      .with(middleware) //<< error occurs here
      .build();
```

output:

```bash
error[E0277]: expected a `Fn<(Request, &'a mut task_local_extensions::extensions::Extensions, Next<'a>)>` closure, found `VCRMiddleware`
  --> framework/tests/acceptance.rs:19:15
   |
19 |         .with(middleware)
   |          ---- ^^^^^^^^^^ expected an `Fn<(Request, &'a mut task_local_extensions::extensions::Extensions, Next<'a>)>` closure, found `VCRMiddleware`
   |          |
   |          required by a bound introduced by this call
   |
   = help: the trait `for<'a> Fn<(Request, &'a mut task_local_extensions::extensions::Extensions, Next<'a>)>` is not implemented for `VCRMiddleware`
   = note: required for `VCRMiddleware` to implement `reqwest_middleware::Middleware`
```

I'm not totally sure what's going on here. The `Middleware` trait still exists in the latest sourcecode for `reqwest-middleware` and is specified in that function, `M` must implement `Middleware`. I'm guessing that the definition of `Middleware` has changed, and this compiler output is a bit misleading.

--- a/Cargo.toml
+++ b/Cargo.toml
@@ -32,7 +32,7 @@ chrono = { version = "0.4", features = ["serde"] }
 http = "0.2"
 lazy_static = "1.4"
 reqwest = { version = "0.11" }
-reqwest-middleware = "0.1.6"
+reqwest-middleware = "0.2.4"
 serde = "1"
 serde_json = "1"
 task-local-extensions = "0.1.4"

